list command results in "Errno 0 (originated from sysctl(KERN_PROCARGS2)" on macOS

The test function test_lis_start() with testcase "Verify output of 'start --help-call'" failed with "Errno 0 (originated from sysctl(KERN_PROCARGS2)" on macOS with Python 3.10 (latest).

Log: https://github.com/pywbem/pywbemtools/actions/runs/21150786886/job/60826939420

```
            if rc != 0:
                if verbose:
                    log_data = get_listener_logdata()
                else:
                    log_data = None
>               raise AssertionError(
                    f"{msg}: The command {cmd_args!r} failed with rc={rc}.\n"
                    f"Situation: {situation}\n"
                    f"Standard output:\n{out}\n"
                    f"Standard error:\n{err}\n"
                    f"{log_data}")
E                   AssertionError: Listing listeners failed: The command ['pywbemlistener', '-o', 'plain', 'list'] failed with rc=1.
E                   Situation: test teardown
E                   Standard output:
E                   
E                   Standard error:
E                   Traceback (most recent call last):
E                     File "/Users/runner/hostedtoolcache/Python/3.10.11/arm64/bin/pywbemlistener", line 7, in <module>
E                       sys.exit(main())
E                     File "/Users/runner/work/pywbemtools/pywbemtools/pywbemtools/pywbemlistener/pywbemlistener.py", line 160, in main
E                       ctx.command.invoke(ctx)  # Calls cli()
E                     File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/core.py", line 1873, in invoke
E                       return _process_result(sub_ctx.command.invoke(sub_ctx))
E                     File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/core.py", line 1269, in invoke
E                       return ctx.invoke(self.callback, **ctx.params)
E                     File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/core.py", line 824, in invoke
E                       return callback(*args, **kwargs)
E                     File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/decorators.py", line 46, in new_func
E                       return f(get_current_context().obj, *args, **kwargs)
E                     File "/Users/runner/work/pywbemtools/pywbemtools/pywbemtools/pywbemlistener/_cmd_listener.py", line 470, in listener_list
E                       context.execute_cmd(lambda: cmd_listener_list(context))
E                     File "/Users/runner/work/pywbemtools/pywbemtools/pywbemtools/pywbemlistener/_context_obj.py", line 177, in execute_cmd
E                       cmd()  # The command function for the pywbemlistener command
E                     File "/Users/runner/work/pywbemtools/pywbemtools/pywbemtools/pywbemlistener/_cmd_listener.py", line 470, in <lambda>
E                       context.execute_cmd(lambda: cmd_listener_list(context))
E                     File "/Users/runner/work/pywbemtools/pywbemtools/pywbemtools/pywbemlistener/_cmd_listener.py", line 1366, in cmd_listener_list
E                       listeners = get_listeners()
E                     File "/Users/runner/work/pywbemtools/pywbemtools/pywbemtools/pywbemlistener/_cmd_listener.py", line 545, in get_listeners
E                       cmdline = p.cmdline()
E                     File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/psutil/__init__.py", line 748, in cmdline
E                       return self._proc.cmdline()
E                     File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/psutil/_psosx.py", line 347, in wrapper
E                       return fun(self, *args, **kwargs)
E                     File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/psutil/_psosx.py", line 405, in cmdline
E                       return cext.proc_cmdline(self.pid)
E                   OSError: [Errno 0] Undefined error: 0 (originated from sysctl(KERN_PROCARGS2))
E                   
E                   None

tests/unit/pywbemlistener/cli_test_extensions.py:333: AssertionError
```

--- a/changes/1511.fix.rst
+++ b/changes/1511.fix.rst
@@ -1,0 +1,4 @@
+pywbemlistener: Circumvented the behavior that on macOS where "pywbemlistener list"
+could raise OSError "Errno 0 (originated from sysctl(KERN_PROCARGS2)" when
+retrieving the command line of a process when listing the processes, by
+ignoring that error.

--- a/pywbemtools/pywbemlistener/_cmd_listener.py
+++ b/pywbemtools/pywbemlistener/_cmd_listener.py
@@ -547,6 +547,18 @@ def get_listeners(name=None):
                 psutil.NoSuchProcess):
             # Ignore processes we cannot access or that ended meanwhile
             continue
+        except OSError as exc:
+            if exc.errno == 0 and "KERN_PROCARGS2" in str(exc):
+                # Ignore the following error:
+                #   [Errno 0] Undefined error: 0 (originated from
+                #   sysctl(KERN_PROCARGS2))
+                # See https://github.com/giampaolo/psutil/issues/2708
+                # Note: psutil plans to raise AccessDenied for this error in
+                # the future, see
+                # https://github.com/giampaolo/psutil/issues/2708 .
+                continue
+            raise
+
         for i, item in enumerate(cmdline):
             if item.endswith(cmdname):
                 listener_index = i

